pulp_rpm/app/serializers.py

```diff
@@ -404,12 +435,111 @@ class UpdateRecordSerializer(NoArtifactContentSerializer):
         help_text=_("List of references")
     )
 
+    def create(self, validated_data):
+        """
+        Create UpdateRecord and its subclasses from JSON file.
+
+        Returns:
+            UpdateRecord instance
+
+        """
+        references = validated_data.pop("references", [])
+        pkglist = validated_data.pop("pkglist", [])
+        update_collection_packages_to_save = list()
+        update_references_to_save = list()
+        try:
+            update_record = super().create(validated_data)
+        except IntegrityError:
+            raise serializers.ValidationError("Advisory already exists in Pulp.")
+
+        for collection in pkglist:
+            packages = collection.pop("packages", [])
+            new_pkg = collection
+            new_pkg[PULP_UPDATE_COLLECTION_ATTRS.SHORTNAME] = new_pkg.pop(
+                "short", ""
+            )
+            coll = UpdateCollection(**new_pkg)
+            try:
+                coll.save()
+            except IntegrityError:
+                pass
+            coll.update_record.add(update_record)
+            for package in packages:
+                pkg = UpdateCollectionPackage(**package)
+                pkg.update_collection = coll
+                update_collection_packages_to_save.append(pkg)
+        for reference in references:
+            new_ref = dict()
+            new_ref[PULP_UPDATE_REFERENCE_ATTRS.HREF] = reference.get(
+                CR_UPDATE_REFERENCE_ATTRS.HREF, ""
+            )
+            new_ref[PULP_UPDATE_REFERENCE_ATTRS.ID] = reference.get(
+                CR_UPDATE_REFERENCE_ATTRS.ID, ""
+            )
+            new_ref[PULP_UPDATE_REFERENCE_ATTRS.TITLE] = reference.get(
+                CR_UPDATE_REFERENCE_ATTRS.TITLE, ""
+            )
+            new_ref[PULP_UPDATE_REFERENCE_ATTRS.TYPE] = reference.get(
+                CR_UPDATE_REFERENCE_ATTRS.TYPE, ""
+            )
+            ref = UpdateReference(**new_ref)
+            ref.update_record = update_record
+            update_references_to_save.append(ref)
+
+        if update_collection_packages_to_save:
+            UpdateCollectionPackage.objects.bulk_create(update_collection_packages_to_save)
+        if update_references_to_save:
+            UpdateReference.objects.bulk_create(update_references_to_save)
+
+        return update_record
+
+    def validate(self, data):
+        """
+        Read a file for a JSON data and validate a UpdateRecord data.
+
+        Also change 'from' to 'fromstr' if exists as this is usually handle by
+        createrepo_c which is not used here.
+        """
+        update_record_data = dict()
```

Comment:
you can update `data` dict with loaded json file information and use update_record_data in the same say as you did with `references`, so you do not use pop but get and also you would avoid iterating through the dict that you change on the fly